packages/beacon-node/src/chain/seenCache/seenBlockInput.ts

```diff
@@ -0,0 +1,258 @@
+        this.blockInputs.delete(rootHex);
+      }
+    }
+  }
+
+  prune(rootHex: RootHex): void {
+    let blockInput = this.blockInputs.get(rootHex);
+    let parentRootHex = blockInput?.parentRootHex;
+    while (blockInput) {
+      this.blockInputs.delete(blockInput.blockRootHex);
+      blockInput = this.blockInputs.get(parentRootHex ?? "");
+      parentRootHex = blockInput?.parentRootHex;
+    }
+  }
+
+  getBlockInputByBlock({
+    block,
+    source,
+    seenTimestampSec,
+    peerIdStr,
+  }: SourceMeta & {block: SignedBeaconBlock}): IBlockInput {
+    const blockRoot = this.config.getForkTypes(block.message.slot).BeaconBlock.hashTreeRoot(block.message);
+    const blockRootHex = toHex(blockRoot);
+
+    // TODO(peerDAS): Why is it necessary to static cast this here. All conditional paths result in a valid value so should be defined correctly below
+    let blockInput = this.blockInputs.get(blockRootHex) as IBlockInput;
+    if (!blockInput) {
+      const {forkName, daOutOfRange} = this.buildCommonProps(block.message.slot);
+      if (!isForkPostDeneb(forkName)) {
+        blockInput = BlockInputPreData.createFromBlock({
+          block,
+          blockRootHex,
+          daOutOfRange,
+          forkName,
+          source: {
+            source,
+            seenTimestampSec,
+            peerIdStr,
+          },
+        });
+      }
+      // else if (isForkPostFulu(forkName)) {
+      //   blockInput = new BlockInputColumns.createFromBlock({
+      //     block,
+      //     blockRootHex,
+      //     daOutOfRange,
+      //     forkName,
+      //     custodyColumns: this.custodyConfig.custodyColumns,
+      //     sampledColumns: this.custodyConfig.sampledColumns,
+      //     source: {
+      //       source,
+      //       seenTimestampSec,
+      //       peerIdStr
+      //     }
+      //   })
+      // }
+      else {
+        blockInput = BlockInputBlobs.createFromBlock({
+          block: block as SignedBeaconBlock<ForkBlobsDA>,
+          blockRootHex,
+          daOutOfRange,
+          forkName,
+          source: {
+            source,
+            seenTimestampSec,
+            peerIdStr,
+          },
+        });
+      }
+    }
+
+    if (!blockInput.hasBlock()) {
+      blockInput.addBlock({block, blockRootHex, source: {source, seenTimestampSec, peerIdStr}});
+    } else {
+      this.logger?.debug("Attempt to cache block but is already cached on BlockInput", blockInput.getLogMeta());
+      this.metrics?.seenCache.blockInput.duplicateBlockCount.inc();
+    }
+
+    return blockInput;
+  }
+
+  getBlockInputByBlob(
+    {blobSidecar, source, seenTimestampSec, peerIdStr}: SourceMeta & {blobSidecar: deneb.BlobSidecar},
+    opts: GetByBlobOptions = {}
+  ): IBlockInput {
+    const blockRoot = this.config
+      .getForkTypes(blobSidecar.signedBlockHeader.message.slot)
+      .BeaconBlockHeader.hashTreeRoot(blobSidecar.signedBlockHeader.message);
+    const blockRootHex = toHex(blockRoot);
```

Comment:
```suggestion
    const blockRootHex = toRootHex(blockRoot);
```

as it's more performant, same to other places